bot/exts/halloween/candy_collection.py

```diff
@@ -77,33 +77,27 @@ async def on_reaction_add(self, reaction: discord.Reaction, user: discord.Member
                 await self.reacted_msg_chance(message)
             return
 
-        for react in self.msg_reacted:
-            # check to see if the message id of a message we added a
-            # reaction to is in json file, and if nobody has won/claimed it yet
-            if react['msg_id'] == message.id and react['won'] is False:
-                react['user_reacted'] = user.id
-                react['won'] = True
-                try:
-                    # if they have record/candies in json already it will do this
-                    user_records = self.get_candyinfo[user.id]
-                    if str(reaction.emoji) == '\N{CANDY}':
-                        user_records['record'] += 1
-                    if str(reaction.emoji) == '\N{SKULL}':
-                        if user_records['record'] <= 3:
-                            user_records['record'] = 0
-                            lost = 'all of your'
-                        else:
-                            lost = random.randint(1, 3)
-                            user_records['record'] -= lost
-                        await self.send_spook_msg(message.author, message.channel, lost)
-
-                except KeyError:
-                    # otherwise it will raise KeyError so we need to add them to file
-                    if str(reaction.emoji) == '\N{CANDY}':
-                        print('ok')
-                        d = {"userid": user.id, "record": 1}
-                        self.candy_json['records'].append(d)
-                await self.remove_reactions(reaction)
+        if message.id in self.candy_messages and str(reaction.emoji) == '\N{CANDY}':
+            self.candy_messages.remove(message.id)
+            prev_record = self.candy_records.get(str(message.author.id)) or 0
+            self.candy_records[str(message.author.id)] = prev_record + 1
+
+        elif message.id in self.skull_messages and str(reaction.emoji) == '\N{SKULL}':
+            self.skull_messages.remove(message.id)
+
+            if (prev_record := self.candy_records.get(str(message.author.id))) is not None:
+                lost = min(random.randint(1, 3), prev_record)
+                self.candy_records[str(message.author.id)] = prev_record - lost
+
+                if lost == prev_record:
+                    await self.send_spook_msg(message.author, message.channel, 'all of your')
```

Comment:
the argument `all of your` is a string but the method `send_spook_msg` expects an integer value as the number of candies.
You need to update the function type annotation from being `candies: int` to `candies: Union[int, str]`
Don't forget to import `Union` from `typing`.